TestSingleStoreDistributedQueries is flaky and should be single threaded

This one might be fixed at head. Rebased and rerunning. If it's not, then that means it's not enough to include `@Test(singleThreaded=true),` on the superclass. We have to add it to each subclass explicitly.

```
2024-03-06T19:59:01.7894018Z FATAL: com.facebook.presto.testng.services.ReportMultiThreadedBeforeOrAfterMethod: Failed to process [TestClass name=class com.facebook.presto.plugin.singlestore.TestSingleStoreDistributedQueries]: 
2024-03-06T19:59:01.7898504Z java.lang.RuntimeException: Test class com.facebook.presto.plugin.singlestore.TestSingleStoreDistributedQueries should be annotated as @Test(singleThreaded=true), if it contains mutable state as indicated by public void com.facebook.presto.tests.AbstractTestDistributedQueries.setUp()
2024-03-06T19:59:01.7902398Z 	at com.facebook.presto.testng.services.ReportMultiThreadedBeforeOrAfterMethod.reportMultiThreadedBeforeOrAfterMethod(ReportMultiThreadedBeforeOrAfterMethod.java:74)
2024-03-06T19:59:01.7905222Z 	at com.facebook.presto.testng.services.ReportMultiThreadedBeforeOrAfterMethod.onBeforeClass(ReportMultiThreadedBeforeOrAfterMethod.java:42)
2024-03-06T19:59:01.7907386Z 	at org.testng.internal.invokers.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:167)
2024-03-06T19:59:01.7908962Z 	at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:122)
2024-03-06T19:59:01.7910330Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2024-03-06T19:59:01.7911712Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2024-03-06T19:59:01.7912709Z 	at java.lang.Thread.run(Thread.java:750)
2024-03-06T19:59:01.7913130Z 
2024-03-06T19:59:01.7913298Z JVM will be terminated
2024-03-06T19:59:01.8125318Z 2024-03-06T13:59:01.794-0600 INFO Life cycle stopping...
2024-03-06T19:59:02.1023499Z ##[error]Exception in thread "dispatcher-query-4" java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@30e120f9 rejected from java.util.concurrent.ScheduledThreadPoolExecutor@758e8e92[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 311]
```

--- a/presto-clickhouse/src/test/java/com/facebook/presto/plugin/clickhouse/TestClickHouseDistributedQueries.java
+++ b/presto-clickhouse/src/test/java/com/facebook/presto/plugin/clickhouse/TestClickHouseDistributedQueries.java
@@ -44,6 +44,7 @@ import static org.assertj.core.api.Assertions.assertThat;
 import static org.testng.Assert.assertFalse;
 import static org.testng.Assert.assertTrue;
 
+@Test(singleThreaded = true)
 public class TestClickHouseDistributedQueries
             extends AbstractTestDistributedQueries
 {

--- a/presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueries.java
+++ b/presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueries.java
@@ -34,6 +34,7 @@ import static java.util.stream.Collectors.joining;
 import static org.testng.Assert.assertEquals;
 import static org.testng.Assert.assertTrue;
 
+@Test(singleThreaded = true)
 public class TestHiveDistributedQueries
         extends AbstractTestDistributedQueries
 {

--- a/presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueriesWithExchangeMaterialization.java
+++ b/presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueriesWithExchangeMaterialization.java
@@ -32,6 +32,7 @@ import static io.airlift.tpch.TpchTable.getTables;
 import static org.testng.Assert.assertEquals;
 import static org.testng.Assert.assertThrows;
 
+@Test(singleThreaded = true)
 public class TestHiveDistributedQueriesWithExchangeMaterialization
         extends AbstractTestDistributedQueries
 {

--- a/presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueriesWithOptimizedRepartitioning.java
+++ b/presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueriesWithOptimizedRepartitioning.java
@@ -16,11 +16,13 @@ package com.facebook.presto.hive;
 import com.facebook.presto.testing.QueryRunner;
 import com.facebook.presto.tests.AbstractTestDistributedQueries;
 import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
 
 import java.util.Optional;
 
 import static io.airlift.tpch.TpchTable.getTables;
 
+@Test(singleThreaded = true)
 public class TestHiveDistributedQueriesWithOptimizedRepartitioning
         extends AbstractTestDistributedQueries
 {

--- a/presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueriesWithThriftRpc.java
+++ b/presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueriesWithThriftRpc.java
@@ -16,11 +16,13 @@ package com.facebook.presto.hive;
 import com.facebook.presto.testing.QueryRunner;
 import com.facebook.presto.tests.AbstractTestDistributedQueries;
 import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
 
 import java.util.Optional;
 
 import static io.airlift.tpch.TpchTable.getTables;
 
+@Test(singleThreaded = true)
 public class TestHiveDistributedQueriesWithThriftRpc
         extends AbstractTestDistributedQueries
 {

--- a/presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownDistributedQueries.java
+++ b/presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownDistributedQueries.java
@@ -26,6 +26,7 @@ import static com.google.common.collect.Iterables.getOnlyElement;
 import static io.airlift.tpch.TpchTable.getTables;
 import static org.testng.Assert.assertEquals;
 
+@Test(singleThreaded = true)
 public class TestHivePushdownDistributedQueries
         extends AbstractTestDistributedQueries
 {

--- a/presto-hive/src/test/java/com/facebook/presto/hive/TestParquetDistributedQueries.java
+++ b/presto-hive/src/test/java/com/facebook/presto/hive/TestParquetDistributedQueries.java
@@ -30,6 +30,7 @@ import static com.google.common.collect.Iterables.getOnlyElement;
 import static io.airlift.tpch.TpchTable.getTables;
 import static org.testng.Assert.assertEquals;
 
+@Test(singleThreaded = true)
 public class TestParquetDistributedQueries
         extends AbstractTestDistributedQueries
 {

--- a/presto-iceberg/src/test/java/com/facebook/presto/iceberg/IcebergDistributedTestBase.java
+++ b/presto-iceberg/src/test/java/com/facebook/presto/iceberg/IcebergDistributedTestBase.java
@@ -112,6 +112,7 @@ import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
 import static org.testng.Assert.assertTrue;
 
+@Test(singleThreaded = true)
 public class IcebergDistributedTestBase
         extends AbstractTestDistributedQueries
 {

--- a/presto-mysql/src/test/java/com/facebook/presto/plugin/mysql/TestMySqlDistributedQueries.java
+++ b/presto-mysql/src/test/java/com/facebook/presto/plugin/mysql/TestMySqlDistributedQueries.java
@@ -33,7 +33,7 @@ import static com.facebook.presto.testing.MaterializedResult.resultBuilder;
 import static com.facebook.presto.testing.assertions.Assert.assertEquals;
 import static java.util.concurrent.TimeUnit.SECONDS;
 
-@Test
+@Test(singleThreaded = true)
 public class TestMySqlDistributedQueries
         extends AbstractTestDistributedQueries
 {

--- a/presto-singlestore/src/test/java/com/facebook/presto/plugin/singlestore/TestSingleStoreDistributedQueries.java
+++ b/presto-singlestore/src/test/java/com/facebook/presto/plugin/singlestore/TestSingleStoreDistributedQueries.java
@@ -29,7 +29,7 @@ import static java.util.stream.Collectors.joining;
 import static java.util.stream.IntStream.range;
 import static org.testng.Assert.assertTrue;
 
-@Test
+@Test(singleThreaded = true)
 public class TestSingleStoreDistributedQueries
         extends AbstractTestDistributedQueries
 {

